moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -2039,28 +2039,40 @@ double RobotState::testRelativeJointSpaceJump(const JointModelGroup* group, std:
 double RobotState::testAbsoluteJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
                                               double revolute_threshold, double prismatic_threshold)
 {
-  bool test_revolute = revolute_threshold > 0;
-  bool test_prismatic = revolute_threshold > 0;
+  struct
+  {
+    double limit;
+    bool check;
+  } data[2] = { { revolute_threshold, revolute_threshold > 0.0 }, { prismatic_threshold, prismatic_threshold > 0.0 } };
```

Comment:
this struct is overly-complicated and takes much longer to understand than two simple bools in the original approach. 

- you bundle declaring the struct, creating an array of structs, and initializing them unnecessarily, which takes a long time to parse
- you mix in conditional logic with the initializer 
- there is an unnecessary copy of threshold values